### ISSUE TITLE
Connect Admin Api to the Admin UI

### ISSUE THREAD

**Issue by skoruba** [skoruba/IdentityServer4.Admin#478]
Currently, the Admin UI is connected to services directly from the controllers.

I am considering using Admin Api from Admin UI.

It will required some rewrite, but it should be cleaner solution.

Feedback is welcome. 👍 

Thanks

**Comment by danielmeza** [comment]
That's great! This allow us to reduce the maintenance effort and will propagate new features across all system. Remember to use HttpClientFactory and Polly for reduce HTTP connection issues.

**Comment by skoruba** [comment]
Can you describe your idea with Polly?

**Comment by b0** [comment]
@skoruba count on me when time comes for this rewrite...

Thank you

**Comment by devployment** [comment]
Would that mean in general, staying with MVC or going down the SPA route? If SPA would be an option, any thoughts on the frontend stack? Maybe it would be nice to bring some love to https://aurelia.io/

IMHO they deserve being used by great projects to showcase their stuff. :)

I have not too much experience with Aurelia, but would love to get my hands dirty if the time comes.

**Comment by skoruba** [comment]
I want to work still with MVC. Thanks

**Comment by danielmeza** [comment]
@devployment I have a lot of experience with aurelia, and it is a awesome JS framework, but it is JS. Although since the project is oriented to rest noting stop us to build a different UI stack. Personally I will chose something closed to the .net world like Blazor, but for now MVC is just fine.

**Comment by skoruba** [comment]
I agree with you @danielmeza - thanks for feedback. Maybe I will consider some rewrite but in the future, current solution with MVC is stable. 👍🏼

**Comment by mehyaa** [comment]
In my opinion Admin and Admin Api projects can be merged.

Connecting Admin to Api will cause unnecessary navigation, auth steps. That will cause unnecessary delay and complexity. Merging two projects into one and execute all the business logic in common services will be more simple.

Api endpoints can serve under a logical route base name like "/Api/*".

**Comment by skoruba** [comment]
@mehyaa I agree with you. This merge will required bigger rewrite, so I will work on this issue in Q3 of this year - help with this is really welcome. 👍

**Comment by vicentt** [comment]
Hey @skoruba!
Do you want to merge the projects or use the Api on AdminUI for the moment?

**Comment by raminassar** [comment]
any progress on this?

**Comment by skoruba** [comment]
Unfortunately not, currently postponed, I am working on new features to Duende IdentityServer Admin UI.

**Comment by raminassar** [comment]
@skoruba what is the difference between both?
